### PR TITLE
Hsa1/booktrak

### PR DESCRIPTION
**Pull request by AndrewMuh**
Created bookmark UI

### CODE REVIEW

**Comment by yechs** [comment]
Blocked by https://github.com/WilliamsStudentsOnline/wso-api-client/pull/25
Required NPM deployment and dependency update after the PR is merged

**Review by yechs — APPROVED** [review]

**Review by yechs — APPROVED** [review]
I was only able to skim this very briefly. Generally looks okay, no obvious bugs.
For urgency to deploy the feature, I'll approve this.

Merge after fixing the scope issue and updating the dependencies

File: src/router-permissions.js
Comment (yechs) [reply src/router-permissions.js:15]: > (and don't forget the special booktrak write scope)

BTW you may want to do something special about the write scope, e.g. limiting it to the publish listing page.

Comment (AndrewMuh) [reply src/router-permissions.js]: Fixed scoping

**Review by yechs — APPROVED** [review]
For urgency to deploy the feature, I'll approve this.

There are a few more TODOs communicated out of band. We'll make separate issues later on.

**Pull request merged**
